
m 2e11b93fa0a7f3ac97b894ad0c3ed58bde73d7b4 Mon Sep 17 00:00:00 2001
Message-Id: <2e11b93fa0a7f3ac97b894ad0c3ed58bde73d7b4.1517303991.git.jan.steffens@gmail.com>
From: "Jan Alexander Steffens (heftig)" <jan.steffens@gmail.com>
Date: Tue, 30 Jan 2018 10:14:56 +0100
Subject: [PATCH] collector: Always return a new reference to the singleton

Don't hand out our own reference with the newly created singleton, since
our pointer might become invalid if all the references we gave out were
removed.

Fixes a crash happening when opening a second file chooser dialog in
GEdit, Builder or Firefox. Doesn't trigger in the widget factory,
though.
---
 src/cloudproviderscollector.c | 8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

--- src/cloudproviderscollector.c
+++ src/cloudproviderscollector.c
@@ -319,12 +319,8 @@ cloud_providers_collector_dup_singleton (void)
                  self->cancellable,
                  on_bus_acquired,
                  self);
+    }
 
-      return self;
-    }
-  else
-    {
-      return g_object_ref (self);
-    }
+  return g_object_ref (self);
 }
 
-- 
2.16.1


